### DISCUSSION TITLE
[New Feature] Player Factory Unit Copying

### DISCUSSION THREAD

**Discussion by ajgunsauley** [Phobos-developers/Phobos#958] [Ideas and feedback]
### Description:
When a unit is built from a factory type (Infantry, Vehicle, Naval), produce a copy of that unit from every additional factory of the same type.   This would mimic AI behavior but allow the player to take advantage of the same setup.  Currently this is not possible since Ares implementation doesn't permit cloning on a Factory=yes building.

This request would allow both production and cloning functionality to function out of one structure in the faction arsenal instead of requiring two separate buildings -- one for production and one for cloning.

Excluding AircraftTypes from the request.

### Possible Implementation:

```ini
[BuildingType]
IsCopyFactory=Yes
```

Note that if you had the same unit eligible to be built from a factory with this flag and one without, setting the primary to the CopyFactory will not produce a copy from the building where the flag is not set or = no.  However, setting the primary to the =no building would still produce a copy from the CopyFactory, **unless** Factory.ExplicitOnly conditions exclude the unit being produced at the CopyFactory.

**Comment by ajgunsauley** [comment]
Note as well probably that anything with Cloneable=no should not come out of a CopyFactory since this is basically just another form of cloning.

**Comment by ghost** [comment]
I'd say AllowParallelPlayerQueues=yes in [GlobalControls] is more appropriate, same place as AllowParallelAIQueues=yes (Ares). But this is pretty much cheating, you pay 1 unit and spawn a lot more. Could be fun for a mod though, makes having multiple factories more useful than just for production speed.

A more fair way would be for each available factory to build and spawn one of the queued units in parallel, so you could get 1 tank, 1 IFV, 1 helicopter etc at the same time (if you have the money!), like in StarCraft.  UI must also be updated correctly otherwise it will look buggy and cause exploits (some units build faster than others). I don't really think this is possible though, it's too complicated and ruins the original game design...
